tests/algorithms/test_preference_comparisons.py

```diff
@@ -72,6 +72,21 @@ def agent_trainer(agent, reward_net, venv, rng):
     return preference_comparisons.AgentTrainer(agent, reward_net, venv, rng)
 
 
+def assert_info_arrs_equal(arr1, arr2):  # pragma: no cover
+    for item1, item2 in zip(arr1, arr2):
+        assert isinstance(item1, dict)
+        assert isinstance(item2, dict)
+        for key, val in item1.items():
+            assert key in item2
+            if isinstance(val, dict):
```

Comment:
Do we want to enforce that dicts are only nested one layer deep? If not we could just turn this into a recursive call (it'd also simplify the code)?

Reply:
I think it might make sense to leave as is, so we can enforce that dicts are either unnested or nested one layer deep, so that the test breaking will alert us that the info array contains dicts nested even further? I don't have a strong opinion here though.

Reply:
Yeah on reflection changed this to the recursive call, makes it easier to reason about.